File: _assets/scripts/normalizers/languages_normalize.lua
local function fix_properties(data)
    local output = data["properties"]
    local output_attr = python.as_attrgetter(output)
    for index, resource_pack in python.enumerate(data["defined_in"]) do
        if not output_attr.__contains__(resource_pack) then
            output[resource_pack] = python.builtins.dict()
        end
    end
    return output
end

return function (data)
    local output = python.builtins.dict()
    for index, language in python.enumerate(data) do
        output[language["code"]] = fix_properties(language)
    end
    return output
end

File: _assets/scripts/normalizers/languages_normalize.py
from typing import Any, TypedDict

__all__ = ["languages_normalize"]

language_typed_dict = TypedDict("language_typed_dict", code=str, properties=dict[str,Any], defined_in=list[str])

def fix_properties(data:language_typed_dict) -> dict[str,dict[str,Any]]:
    output:dict[str,dict[str,Any]] = data["properties"]
    output.update(
        (resource_pack, {})
        for resource_pack in data["defined_in"]
        if resource_pack not in output
    )
    return output

def languages_normalize(data:list[language_typed_dict]) -> dict[str,dict[str,dict[str,Any]]]:
    return {language["code"]: fix_properties(language) for language in data}
